crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -484,6 +487,7 @@ where
 
     #[inline]
     fn initialize(&mut self, world: &mut World) {
+        self.system_meta.last_change_tick = world.last_change_tick;
         self.param_state = Some(<Param::Fetch as SystemParamState>::init(
             world,
             &mut self.system_meta,
```

Comment:
Leaving the `last_change_tick` set to `0` was probably not correct. From what I could tell, initializing a system "late" would cause it to miss changes in its first `run`, since `last_change_tick` (which, again, was `0`) would be piped directly into any system params and not updated until after the data was fetched once.